client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { getSite } from 'state/sites/selectors';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import QuerySites from 'components/data/query-sites';
+import QuerySitePlans from 'components/data/query-site-plans';
+import { plansList } from 'lib/plans/constants';
+import formatCurrency from 'lib/format-currency';
+
+class PlanThankYouCard extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired
+	};
+
+	render() {
+		const {
+			plan,
+			site,
+			translate,
+			siteId
+		} = this.props;
+		// Non standard gridicon sizes are used here because we use them as background pattern with various sizes and rotation
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return (
+			<div className="plan-thank-you-card">
+				<QuerySites siteId={ siteId } />
```

Comment:
If `siteId` is a falsey value, this would query all sites of a user. Render `<QuerySites>` only if `siteId` is truthy, like:

```js
{ siteId && <QuerySites siteId={ ... } /> }
```